GPU/Common/VertexShaderGenerator.cpp

```diff
@@ -1056,7 +1062,7 @@ bool GenerateVertexShader(const VShaderID &id, char *buffer, const ShaderLanguag
 						break;
 					case GE_PROJMAP_NORMALIZED_NORMAL:  // Use normalized transformed normal as source
 						if (hasNormal)
-							temp_tc = flipNormal ? "vec4(normalize(-normal), 1.0)" : "vec4(normalize(normal), 1.0)";
+							temp_tc = StringFromFormat("length(%snormal) == 0.0 ? vec4(0.0, 0.0, 1.0, 1.0) : vec4(normalize(%snormal), 1.0)", flipNormal ? "-" : "");
```

Comment:
@unknownbrackets I think the bug is here. Two %s but one parameter.

You don't really need the sign in the length check anyway...

Reply:
Oops, I don't think I intended to have it twice.  Bah.

-[Unknown]

Reply:
It's kinda ridiculous that this compiles.

Reply:
Makes me miss D where [this wouldn't](https://dlang.org/library/std/stdio/writef.html).  I'm sure rust has something similar.

-[Unknown]

Reply:
Indeed, this would break with Rust formatting macros.